ecs-agent/logger/log.go

```diff
@@ -236,62 +235,115 @@ func setInstanceLevelDefault() string {
 	return DEFAULT_LOGLEVEL
 }
 
+// SetInstanceLogLevel explicitly sets the log level for instance logs.
+func SetInstanceLogLevel(instanceLogLevel string) {
+	parsedLevel, ok := logLevels[strings.ToLower(instanceLogLevel)]
+	if ok {
+		Config.lock.Lock()
+		defer Config.lock.Unlock()
+		Config.instanceLevel = parsedLevel
+		reloadConfig()
+	} else {
+		seelog.Error("Instance log level mapping not found")
+	}
+}
+
+// SetDriverLogLevel explicitly sets the log level for a custom driver.
+func SetDriverLogLevel(driverLogLevel string) {
+	parsedLevel, ok := logLevels[strings.ToLower(driverLogLevel)]
+	if ok {
+		Config.lock.Lock()
+		defer Config.lock.Unlock()
+		Config.driverLevel = parsedLevel
+		reloadConfig()
+	} else {
+		seelog.Error("Driver log level mapping not found")
+	}
+}
+
 // SetConfigLogFile sets the default output file of the logger.
 func SetConfigLogFile(logFile string) {
-	Config.lock.Lock()
-	defer Config.lock.Unlock()
+	if logFile != "" {
+		Config.lock.Lock()
+		defer Config.lock.Unlock()
 
-	Config.logfile = logFile
+		Config.logfile = logFile
+		reloadConfig()
+	}
 }
 
-// SetConfigLogFormat sets the output format of the logger.
+// SetConfigOutputFormat sets the output format of the logger.
 // e.g. json, xml, etc.
-func SetConfigLogFormat(logFormat string) {
+func SetConfigOutputFormat(outputFormat string) {
 	Config.lock.Lock()
 	defer Config.lock.Unlock()
 
-	os.Setenv(LOG_OUTPUT_FORMAT_ENV_VAR, logFormat)
-	Config.outputFormat = logFormat
+	Config.outputFormat = outputFormat
+	reloadConfig()
 }
 
 // SetConfigMaxFileSizeMB sets the max file size of a log file
 // in Megabytes before the logger rotates to a new file.
 func SetConfigMaxFileSizeMB(maxSizeInMB float64) {
-	Config.lock.Lock()
-	defer Config.lock.Unlock()
+	if maxSizeInMB > 0 {
+		Config.lock.Lock()
+		defer Config.lock.Unlock()
+
+		Config.MaxFileSizeMB = maxSizeInMB
+		reloadConfig()
+	}
+}
 
-	// Parse unit as string to set as environment variable.
-	strsize := strconv.FormatFloat(maxSizeInMB, 'f', -1, 64)
-	os.Setenv(LOG_MAX_FILE_SIZE_ENV_VAR, strsize)
-	Config.MaxFileSizeMB = maxSizeInMB
+// SetRolloverType sets the logging rollover constraint.
+// This should be either size or date. Logger will roll
+// to a new log file based on this constraint.
+func SetRolloverType(rolloverType string) {
+	if rolloverType == "date" || rolloverType == "size" {
```

Comment:
"none" should be accepted too.

Reply:
Can we please add unit tests for the setters to make sure our validation works as expected?

Reply:
good catch, will update with some more unit tests